### PR TITLE
Add Nova UX for trip approval

### PR DESCRIPTION
**Pull request by kberzinch**
- Nova UX work for #4199 
- Partial progress towards #4210

### CODE REVIEW

File: app/Nova/TravelAssignment.php [hunk 1]
```diff
@@ -105,7 +106,17 @@ public function fields(NovaRequest $request): array
 
             BelongsTo::make('Trip', 'travel', Travel::class)
                 ->withoutTrashed()
-                ->rules('required', 'unique:travel_assignments,travel_id,NULL,id,user_id,'.$request->user)
+                ->rules(
+                    'required',
+                    'unique:travel_assignments,travel_id,NULL,id,user_id,'.$request->user,
```
Comment (JoshOldenburg) [inline app/Nova/TravelAssignment.php:111]: Is there supposed to be a `->id` here?

File: app/Nova/Travel.php [hunk 2]
```diff
@@ -512,19 +569,81 @@ static function (BooleanGroup $field, NovaRequest $request, FormData $formData):
...
+                    $assignment->envelope->reduce(
+                        static fn (bool $carry, \App\Models\DocuSignEnvelope $envelope): bool => $carry &&
+                            $envelope->summary_filename !== null &&
+                            // @phan-suppress-next-line PhanPossiblyNullTypeArgumentInternal
+                            file_exists($envelope->summary_filename),
+                        true
+                    ),
+                true
+            )
+        ) {
```
Comment (JoshOldenburg) [inline app/Nova/Travel.php:600]: Some comments would help here.

**Review by JoshOldenburg — CHANGES_REQUESTED** [review]
I'd suggest saving the user who approved it, though maybe the nova action log is enough? A couple minor things, but overall it looks good.

File: app/Policies/TravelPolicy.php [hunk 3]
```diff
@@ -41,35 +41,56 @@ public function create(User $user): bool
...
+    public function addTravelAssignment(User $user, Travel $travel): bool
+    {
+        return $user->hasRole('admin') || (
+            $travel->status === 'draft' &&
+            (
+                $user->can('manage-travel') ||
+                $travel->primaryContact === $user
+            )
+        );
+    }
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply app/Policies/TravelPolicy.php:95]: That's covered by `TravelAssignmentPolicy::delete`.

This is here separately to hide the create button on trip detail.

File: app/Nova/TravelAssignment.php (hunk 1 above)
Comment (kberzinch) [reply app/Nova/TravelAssignment.php:111]: No, `user` here is referring to a request attribute, not the authenticated user.

**Review by JoshOldenburg — APPROVED** [review]

File: app/Nova/Travel.php [hunk 4]
```diff
@@ -512,19 +569,82 @@ static function (BooleanGroup $field, NovaRequest $request, FormData $formData):
...
+                    ->withoutActionEvents()
+                    ->canRun(static fn (): bool => true);
+            } elseif ($trip->actions()->where('user_id', '=', $request->user()->id)->exists()) {
+                $actions[] = Action::danger(
+                    ReviewTrip::make()->name(),
+                    'You can\'t review this trip because you have made changes to it.'
+                )
+                    ->withoutConfirmation()
+                    ->withoutActionEvents()
+                    ->canRun(static fn (): bool => true);
```
Comment (evan10s) [inline app/Nova/Travel.php:641]: this implies that if there are minor issues to fix, the reviewer can't just fix them, which might be a mildly frustrating thing

**Review by evan10s — COMMENTED** [review]
would like to see the actual UX in apiary-test before approving

File: app/Nova/Travel.php (hunk 4 above)
Comment (kberzinch) [reply app/Nova/Travel.php:641]: Yeah, that's intentional. I'm going for a separation of duties here.

**Review by evan10s — APPROVED** [review]

**Pull request merged**